### PR TITLE
Backport timer `reset()` fix

### LINKED ISSUE #242: Backport timer `reset()` fix to v0.16
### What happened?

This issue:

* #241

### What did you expect instead?

The timer resetting properly.

### Affected version(s)

v0.15+

### Affected part(s)

Utility receivers, `Event`, `FileWatcher`, `Timer`, etc. (part:receivers)

### Extra information

We need this backported to v0.16 becasue the SDK v0.25 uses it.

### PR DESCRIPTION
**Pull request by llucax**
- Clear release notes
- Add test for `reset()` while a timer is being waited on
- Fix reseting the timer while it is being waited on
- Update the release notes
- Fix typo in comment

Fixes #242.

### CODE REVIEW

File: src/frequenz/channels/util/_timer.py [hunk 1]
```diff
@@ -652,10 +652,15 @@ async def ready(self) -> bool:
 
         now = self._now()
         time_to_next_tick = self._next_tick_time - now
+
         # If we didn't reach the tick yet, sleep until we do.
-        if time_to_next_tick > 0:
+        # We need to do this in a loop also reacting to the reset event, as the timer
+        # could be reset while we are sleeping, in which case we need to recalculated
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/channels/util/_timer.py]: I thought this was amended in the original patch :(
```suggestion
        # could be reset while we are sleeping, in which case we need to recalculate
```

**Review by daniel-zullo-frequenz — COMMENTED** [review]
LGTM, I'm happy to approve it as there is only a typo, please just let me know

File: src/frequenz/channels/util/_timer.py (hunk 1 above)
Comment (llucax) [reply src/frequenz/channels/util/_timer.py]: Damn, this is in the original PR too #241.

**Comment by llucax** [comment]
OK, I accepted the suggestion, now we need to forwardport the fix of the backported fix :rofl:

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**